subsys/net/ip/net_core.c

```diff
@@ -32,6 +32,16 @@ LOG_MODULE_REGISTER(net_core, CONFIG_NET_CORE_LOG_LEVEL);
 #include <net/lldp.h>
 #endif
 
+#if defined(CONFIG_COAP)
+extern void net_coap_init(void);
+#else
+static inline void net_coap_init(void)
+{
+	return;
+}
+#endif
+
+
```

Comment:
Please add this to coap.h, and include coap.h in this file. We can also call it coap_init(), the net_ prefix is not really needed in this case (as the coap API does not have it).

Reply:
Also, should we add a check to coap_init() that it cannot be initialized multiple times?

Reply:
That's the reason I didn't include it in the header -- because it's not supposed to be publicly exposed - that would probably mean people peruse `coap.h` and infer that THE APP should be calling `coap_init()`. I felt declaring it `extern` was preferable to exposing internal stuff in public headers. This way, only `net_core` even knows it exists.

Reply:
(this would also mean the multiple calls check would be superfluous)

Reply:
> hat would probably mean people peruse `coap.h` and infer

Sure, that is certainly a valid concern. We can also place the init declaration to `subsys/net/ip/net_private.h` as there already other init function declarations.

Reply:
Yeah that sounds better. How's this?